### ISSUE TITLE
How to force Reader to skip empty first line in windows?

### ISSUE THREAD

**Issue by dariusheivazi** [thephpleague/csv#393]
_(Fill in the relevant information below to help triage your issue.)_

|    Q        |   A
|------------ | ------
| Version     | 9.6

### Question
How to force Reader to skip empty first line in windows?

_(Please explain in plain english your question)_

When reading files with windows EOL (CR LF), the Reader class, makes a record with the following var_dump result as the fist record:
The empty.csv is a empty file with one EOL (CR LF) in it.
FILE CONTENTS:
1: CR LF
2:

CODE: 
        $reader = Reader::createFromPath('i:\empty.csv');
        $reader->skipEmptyRecords();
        $records = iterator_to_array($reader);
        var_dump($records);

OUTPUT:
array(1) {
  [0] =>
  array(1) {
    [0] =>
    string(0) ""
  }
}

However next empty lines are skipped as expected. e.g. If I change the file contents to:
FILE CONTENTS:
1: CR LF
2: foo,bar CR LF
3: CR LF
4:

out put will be:
OUTPUT:
array(2) {
  [0] =>
  array(1) {
    [0] =>
    string(0) ""
  }
  [1] =>
  array(2) {
    [0] =>
    string(3) "foo"
    [1] =>
    string(3) "bar"
  }
}

As you see, the 3rd row is skipped but the first empty row problem is still there.
I have to deal with a lot of small csv files, all build in windows and the empty.csv is a typical among them. By now I workaround the problem by checking the file size, but when it comes to files with several lines, I have to check first lines every time I fetch the CSV. I'm sure there is a better way. 
Please help. 
Thanks.

**Comment by nyamsprod** [comment]
@dariusheivazi thanks for using the library. Did you try to use the `Statement` object as explain in the documentation ? I'm afraid I can only point you to that direction as it does not seems to be a bug or an undocumented feature

**Comment by dariusheivazi** [comment]
> @dariusheivazi thanks for using the library. Did you try to use the `Statement` object as explain in the documentation ? I'm afraid I can only point you to that direction as it does not seems to be a bug or an undocumented feature

@nyamsprod, Thanks for your response. Actually I use Statement Objects in other parts of the the application. But for this particular problem, (I mean when the first empty line is not skipped), I thought the problem could be solved by a way more straight than using Statement objects.

**Comment by nyamsprod** [comment]
@dariusheivazi I'm guilty of naming stuff poorly 😉   `Reader::skipEmptyRecords` is the default. So records deemed empty are already removed. 

In you case I think the first line is technically not empty so the only way to remove it is to filter it out using the `Statement` object. At least that's what I understood from your explanation.

**Comment by dariusheivazi** [comment]
> @dariusheivazi I'm guilty of naming stuff poorly 😉 `Reader::skipEmptyRecords` is the default. So records deemed empty are already removed.
> 
> In you case I think the first line is technically not empty so the only way to remove it is to filter it out using the `Statement` object. At least that's what I understood from your explanation.


HI, Really I dont know which part of my words were confusing. The library does not skip empty recoreds as it is promised in documentation: **By default the CSV document normalization removes empty records**. I investigated the flow. The problem is the normalizing function. If the first record is empty, it does not skip it. Yes. you are right, technically first line may contain BOM, so the normalizing funtion does not found it as **null**. The resolution is: make it a little bit more smart:  Just change the Line 255 of file Reader.php:
FROM
`return is_array($record) && ($this->is_empty_records_included || $record != [null]);
`
TO:
`return is_array($record) && ($this->is_empty_records_included || ($record != [null] && $record != [$this->input_bom]));`

and It should work, of course if this does not cuase BC break. Anyway. Thanks for this great library.
